staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/dynamic_serving_content.go

```diff
@@ -124,31 +126,34 @@ func (c *DynamicCertKeyPairContent) loadCertKeyPair() error {
 
 // RunOnce runs a single sync loop
 func (c *DynamicCertKeyPairContent) RunOnce(ctx context.Context) error {
-	return c.loadCertKeyPair()
+	return c.loadCertKeyPair(ctx)
 }
 
 // Run starts the controller and blocks until context is killed.
 func (c *DynamicCertKeyPairContent) Run(ctx context.Context, workers int) {
 	defer utilruntime.HandleCrash()
 	defer c.queue.ShutDown()
 
-	klog.InfoS("Starting controller", "name", c.name)
-	defer klog.InfoS("Shutting down controller", "name", c.name)
+	logger := klog.FromContext(ctx)
+	logger.Info("Starting controller", "name", c.name)
+	defer logger.Info("Shutting down controller", "name", c.name)
 
 	// doesn't matter what workers say, only start one.
-	go wait.Until(c.runWorker, time.Second, ctx.Done())
+	go wait.UntilWithContext(ctx, func(ctx context.Context) {
+		c.runWorker(ctx)
+	}, time.Second)
 
 	// start the loop that watches the cert and key files until stopCh is closed.
 	go wait.Until(func() {
-		if err := c.watchCertKeyFile(ctx.Done()); err != nil {
-			klog.ErrorS(err, "Failed to watch cert and key file, will retry later")
+		if err := c.watchCertKeyFile(ctx, ctx.Done()); err != nil {
```

Comment:
Use `wait.UntilWithContext` instead of `wait.Until + ctx.Done()`.

Reply:
ok